src/context/TodoProvider.tsx

```diff
@@ -0,0 +1,93 @@
+import {
+  FC, ReactNode, createContext, useContext, useEffect, useMemo, useState,
+} from 'react';
+import { Todo } from '../types/Todo';
+import { getTodos } from '../api/todos';
+
+  type Props = {
+    children: ReactNode,
+  };
+
+  type TodoProviderT = {
+    todos: Todo[],
+    setTodos: (React.Dispatch<React.SetStateAction<Todo[] | any>>),
+    errorMessage: string,
+    setErrorMessage: React.Dispatch<React.SetStateAction<string>>,
+    option: string,
+    setOption: React.Dispatch<React.SetStateAction<string>>,
+    visibleTodos: Todo[],
+    tempTodo: Todo | null,
+    setTempTodo: React.Dispatch<React.SetStateAction<Todo | null>>,
+    uncompletedCounter: number,
+  };
+
+const TodoContext = createContext<TodoProviderT>({
+  todos: [],
+  setTodos: () => {},
+  errorMessage: '',
+  setErrorMessage: () => {},
+  option: '',
+  setOption: () => {},
+  visibleTodos: [],
+  tempTodo: null,
+  setTempTodo: () => {},
+  uncompletedCounter: 0,
+});
+
+const TodoProvider: FC<Props> = ({ children }) => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [errorMessage, setErrorMessage] = useState<string>('');
+  const [option, setOption] = useState<string>('all');
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [uncompletedCounter, setUncompletedCounter] = useState<number>(0);
+
+  useEffect(() => {
+    setErrorMessage('');
+    getTodos(12121)
+      .then(data => setTodos(data))
+      .catch(() => setErrorMessage('Unable to load todos'));
+    setTimeout(() => setErrorMessage(''), 3000);
```

Comment:
this 3sec delay should be implemented in Error Component to keep his functionality together